packages/xo-server/src/index.js

```diff
@@ -381,18 +384,37 @@ async function makeWebServerListen(
     ...opts
   }
 ) {
-  if (cert && key) {
-    ;[opts.cert, opts.key] = await Promise.all([readFile(cert), readFile(key)])
-    if (opts.key.includes('ENCRYPTED')) {
-      opts.passphrase = await new Promise(resolve => {
-        // eslint-disable-next-line no-console
-        console.log('Encrypted key %s', key)
-        process.stdout.write(`Enter pass phrase: `)
-        pw(resolve)
-      })
-    }
-  }
   try {
+    if (cert && key) {
+      try {
+        ;[opts.cert, opts.key] = await Promise.all([
+          readFile(cert),
+          readFile(key),
+        ])
+        if (opts.key.includes('ENCRYPTED')) {
+          opts.passphrase = await new Promise(resolve => {
+            // eslint-disable-next-line no-console
+            console.log('Encrypted key %s', key)
+            process.stdout.write(`Enter pass phrase: `)
+            pw(resolve)
+          })
+        }
+      } catch (error) {
+        if (!(autoCert && error.code === 'ENOENT')) {
+          throw error
+        }
+
+        const pems = await genSelfSignedCert()
+        await Promise.all([
+          outputFile(cert, pems.cert, { flag: 'wx', mode: 0o400 }),
+          outputFile(key, pems.key, { flag: 'wx', mode: 0o400 }),
```

Comment:
IMHO, we should be conservative for now and never override an existing file.

If one of the file exists, it will fail to write the new content and HTTPS will not be set up, subsequent run will also fail because the cert and the key won't match, it will be up to the user to properly remove both these files.

FYI, in the future, I'd also like to automatically handle expired self cert.

Reply:
Ok, I just read about the `x` flag. Sounds good, then.